box2dbody.cpp

```diff
@@ -282,15 +307,21 @@ void Box2DBody::synchronize()
 {
     Q_ASSERT(mBody);
 
-    if (sync(mBodyDef.position, mBody->GetPosition())) {
-        if (mTarget)
-            mTarget->setPosition(mWorld->toPixels(mBodyDef.position));
-        emit positionChanged();
-    }
-
     if (sync(mBodyDef.angle, mBody->GetAngle()))
         if (mTarget)
             mTarget->setRotation(toDegrees(mBodyDef.angle));
+
+    if (sync(mBodyDef.position, mBody->GetPosition())) {
+        if (mTarget) {
+            mTarget->setPosition(
+                        mTarget->transformOrigin() == QQuickItem::TopLeft ?
+                            mWorld->toPixels(mBodyDef.position) :
+                            originOffset(true));
+
+        }
+    }
+    emit positionChanged();
```

Comment:
Why is `positionChanged` now emitted all the time, and why did you reverse the order of updating position and angle?